chapters/common_pitfalls.md

```diff
@@ -0,0 +1,44 @@
+# Common Pitfalls for new Vulkan Developers
+
+This is a short list of common assumptions and traps developers new to Vulkan can make.  
```

Comment:
One thing that would be good to add here is a piece of general advice that you don't need to aim for 100% perfection - you have to judge what's good enough for your use case. Code in the way that feels natural, follow this guide, watch out for pitfalls, and only if you start seeing poor performance you should use perf recommendations and PGO to figure out how to improve - don't twist yourself in knots to get things working a particular way unless you need to.

Reply:
(Noting that this is a common theme for this whole section)

Reply:
I opted to put many of the points you made into the "Vulkan is a tool box" section, since it had a lot of overlap over how to create a well rounded solution.